streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -294,11 +306,13 @@ public synchronized void start() {
         log.debug("Starting Kafka Stream process");
 
         if (state == KafkaStreams.State.CREATED) {
-            for (final StreamThread thread : threads)
-                thread.start();
-
             setState(KafkaStreams.State.RUNNING);
 
+            for (final StreamThread thread : threads) {
```

Comment:
@mjsax btw I don't want to appear as dismissive of your point, it's a good one. Elaborating further, all streams start in RUNNING, so there is some logic in having Kafka Streams also start as RUNNING. Otherwise there is a whole bunch of logic that has to be rethought and it's not worth the effort IMO. Thanks.